File: RF-Agent-MATLAB/tasks/matlab_task/original_reward.m
function reward = reward_fcn(state, u_action, prev_u_action, flag)
% Replace this starter reward with your current MATLAB baseline reward.
positionTerm = -abs(getfield_default(state, 'position_error', 0.0));
velocityTerm = -0.1 * abs(getfield_default(state, 'velocity_error', 0.0));
controlTerm = -0.001 * sum(u_action(:).^2);
successBonus = 5.0 * getfield_default(state, 'success', 0.0);
constraintTerm = -10.0 * getfield_default(state, 'constraint_violation', 0.0);
reward = positionTerm + velocityTerm + controlTerm + successBonus + constraintTerm;
end

function value = getfield_default(data, name, defaultValue)
if isstruct(data) && isfield(data, name)
    value = data.(name);
else
    value = defaultValue;
end
end

File: RF-Agent-Python/tasks/python_task/original_reward.py
def compute_reward(obs, action, next_obs, info):
    """Baseline reward placeholder.

    Replace this file with your original reward implementation. RF-Agent-Python
    includes this code in the LLM prompt as a reference, but generated candidates
    are saved separately as candidate_*/reward_fcn.py.
    """
    position_term = -abs(next_obs["position_error"])
    velocity_term = -0.1 * abs(next_obs["velocity_error"])
    torque = action["motor_torque"]
    control_term = -0.001 * sum(t * t for t in torque)
    constraint_term = -10.0 * info["constraint_violation"]
    success_bonus = 5.0 * float(info["success"])
    return position_term + velocity_term + control_term + constraint_term + success_bonus

